modules/jaggery-apps/portal/util/utility.jag

```diff
@@ -95,24 +43,20 @@ function setServerURL() {
             }
         }
 
-        //create server service url
-        serverUrl = identityServerURL.getServerURL("", true, true) + "/services";
-
-        //basic server core url : https://<HOST_NAME>:<PORT>/<PROXY_CONTEXT_PATH>
-        var serverCoreUrl = identityServerURL.getServerURL("", true, false);
+        if ( serverOrigin == null || serverOrigin === '' || serverOrigin === 'null' || serverOrigin.length <= 0 ){
+            serverCoreUrl = identityServerURL.getServerURL("", true, false);
+            serverUrl = identityServerURL.getServerURL("services", true, true);
+        } else {
+            serverCoreUrl = serverOrigin + proxyContextPath;
+            serverUrl = serverCoreUrl + "/services";
+        }
```

Comment:
IMO, unless severOrigin which you have added not configured you can build from the IdentityUtil. That handles proxy context paths anyway.